### PR TITLE
Support for labels and creating branch

### PR DESCRIPTION
**Pull request by jarradseers**
Great library, just adding more functionality as I require, this supports the listing, creation, updating and deletion of project labels.

I've also added the ability to create a project branch.

I started looking at writing unit tests but there were others failing ;)
Also updated the README with info.

### CODE REVIEW

**Comment by jarradseers** [comment]
Please can we merge this? i'd like to use it in production via npm :) cheers

**Comment by fengmk2** [comment]
Nice job, I will review this soon!

File: lib/resources/repository.js [hunk 1]
```diff
@@ -37,6 +37,21 @@ Repository.prototype.getBranches = function (params, callback) {
...
+ * Create a repository branch.
+ * still returns a 200 Ok status code.
+ *
+ * @param {Object} params
+ *  - {String} id The ID of a project
+ *  - {String} branch_name The name of the branch
+ *  - {String} ref Create branch from commit SHA or existing branch
+ * @param {Function} callback
+ */
+Repository.prototype.createBranch = function (params, callback) {
```
Comment (fengmk2) [inline lib/resources/repository.js]: This api has exists on https://github.com/repo-utils/gitlab/blob/master/lib/resources/repository_branch.js#L22

**Comment by fengmk2** [comment]
I will merge this manual.

File: lib/resources/repository.js (hunk 1 above)
Comment (fengmk2) [inline lib/resources/repository.js]: I removed this createBranch function.

Comment (fengmk2) [inline lib/resources/repository.js]: Use this instead

#### client.repositoryBranches.create({id, branch_name, ref})

Create a repository branch on a project.

Parameters:

- id (required) - The ID or NAMESPACE/PROJECT_NAME of a project
- branch_name (required) - The name of the branch.
- ref (required) - Create branch from commit SHA or existing branch.

**Comment by fengmk2** [comment]
landed 9d457be82520445c3f96f01eaa25c2a776b212b4

**Comment by fengmk2** [comment]
node-gitlab@1.2.0 published on npm now.

**Comment by jarradseers** [comment]
Awesome, thanks for this, sorry I didn't realise that the create branch functionality existed :) Cheers!

**Pull request closed without merging**